Report/PortfolioLooper/PortfolioLooper.cs

```diff
@@ -323,13 +324,24 @@ private IEnumerable<PointInTimePortfolio> ProcessOrders(IEnumerable<Order> order
                 Algorithm.SetDateTime(order.Time);
 
                 var orderSecurity = Algorithm.Securities[order.Symbol];
-                if (order.LastFillTime == null)
+                DateTime lastFillTime;
+
+                if ((order.Type == OrderType.MarketOnOpen || order.Type == OrderType.MarketOnClose) &&
+                    (order.Status == OrderStatus.Filled || order.Status == OrderStatus.PartiallyFilled) && order.LastFillTime == null)
+                {
+                    lastFillTime = order.Time;
+                }
+                else if (order.LastFillTime == null)
                 {
                     Log.Trace($"Order with ID: {order.Id} has been skipped because of null LastFillTime");
                     continue;
                 }
+                else
+                {
+                    lastFillTime = order.Time;
```

Comment:
Seems this else should be using `LastFillTime`? Also would be nice to see more tests for the `PortfolioLooper`